Assets/MRTK/Core/Utilities/Physics/MixedRealityRaycaster.cs

```diff
@@ -120,91 +136,130 @@ public static bool RaycastSpherePhysicsStep(RayStep step, float radius, LayerMas
-                : TryGetPrioritizedPhysicsHit(UnityEngine.Physics.SphereCastAll(step.Origin, radius, step.Direction, maxDistance, UnityEngine.Physics.AllLayers), prioritizedLayerMasks, focusIndividualCompoundCollider, out physicsHit);
+            using (RaycastSpherePhysicsStepPerfMarker.Auto())
+            {
+                bool result = false;
+                if (prioritizedLayerMasks.Length == 1)
+                {
+                    // If there is only one priority, don't prioritize
+                    result = UnityEngine.Physics.SphereCast(step.Origin, radius, step.Direction, out physicsHit, maxDistance, prioritizedLayerMasks[0]);
+                }
+                else
+                {
+                    // Raycast across all layers and prioritize
+                    int hitCount = UnityEngine.Physics.SphereCastNonAlloc(step.Origin, radius, step.Direction, SphereCastHits, maxDistance, UnityEngine.Physics.AllLayers);
+                    result = TryGetPrioritizedPhysicsHit(SphereCastHits, hitCount, prioritizedLayerMasks, focusIndividualCompoundCollider, out physicsHit);
+                }
 
-            Profiler.EndSample(); // RaycastSpherePhysicsStep
-            return result;
+                return result;
+            }
         }
 
         /// <summary>
         /// Tries to get the prioritized physics raycast hit based on the prioritized layer masks.
         /// </summary>
         /// <remarks>Sorts all hit objects first by layerMask, then by distance.</remarks>
         /// <returns>The minimum distance hit within the first layer that has hits.</returns>
-        public static bool TryGetPrioritizedPhysicsHit(RaycastHit[] hits, LayerMask[] priorityLayers, bool focusIndividualCompoundCollider, out RaycastHit raycastHit)
+        public static bool TryGetPrioritizedPhysicsHit(
+            RaycastHit[] hits,
+            LayerMask[] priorityLayers,
+            bool focusIndividualCompoundCollider,
+            out RaycastHit raycastHit)
         {
-            Profiler.BeginSample("[MRTK] MixedRealityRaycaster.TryGetPrioritizedPhysicsHit");
+            return TryGetPrioritizedPhysicsHit(
+                hits,
+                hits.Length,
+                priorityLayers,
+                focusIndividualCompoundCollider,
+                out raycastHit);
+        }
 
-            raycastHit = default(RaycastHit);
+        private static readonly ProfilerMarker TryGetPrioritizedPhysicsHitPerfMarker = new ProfilerMarker("[MRTK] MixedRealityRaycaster.TryGetPrioritizedPhysicsHit");
 
-            if (hits.Length == 0)
+        /// <summary>
+        /// Tries to get the prioritized physics raycast hit based on the prioritized layer masks.
+        /// </summary>
+        /// <remarks>Sorts all hit objects first by layerMask, then by distance.</remarks>
+        /// <returns>The minimum distance hit within the first layer that has hits.</returns>
+        public static bool TryGetPrioritizedPhysicsHit(
```

Comment:
I know the previous implementation is already `public` but...do you think it needs to be? These feel like implementations internal to the raycaster.

Reply:
reducing visibility is a breaking change. i wanted to avoid breaking anything,

Reply:
This one's a new method signature though, right?

Reply:
true. the intent was to enable the nonalloc form of raycast to be used in place of the original. do you think we should make this one private?

Reply:
I'm fine with either. My gut says to start `private` and expose if needed

Reply:
done